[Shi Hui Ling] iP

porkeypine opened the pull request.

Commit:
Add Gradle support

porkeypine committed:
Level 1 - implemented greet, echo and exit

porkeypine committed:
Level 2 - implemented add and list tasks

porkeypine committed:
Level 2 - implemented add and list tasks

porkeypine committed:
Level 3 - implemented mark as done

porkeypine committed:
Level 3 - implemented mark as done

porkeypine committed:
Level 3 - implemented mark as done

porkeypine committed:
Level 4 - implemented todo, event and deadline

porkeypine committed:
Automated Text UI Testing

porkeypine committed:
Level-5 implemented error handling with exceptions

porkeypine committed:
Level 6 - implemented delete

porkeypine committed:
A-Enums - implemented enums for Duke's character lines

porkeypine committed:
Level 7

fanminj commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,120 @@
+        INVALIDINPUT("Invalid input, dummy."),
+        INVALIDCOMMAND("I don't understand what you mean.");
+
+        private final String line;
```
Perhaps the variable name could be a little more intuitive?

fanminj commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,120 @@
+import static java.lang.Integer.parseInt;
+
 public class Duke {
+    private enum Speaking {
```
Maybe a noun would be better here?

tortillatiff replied:
Is giving your enum a noun name a better idea?

fanminj reviewed (commented):
Good attempt! Perhaps you can extract out some logic from the Duke class so it would be easier for others to read your code.

tortillatiff commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,120 @@
+import main.java.*;
```
Should the classes in main.java be imported explicitly instead of a wildcard import?

MerlinLim replied:
Yes, he should use single class imports instead of wildcards

tortillatiff reviewed (commented):
Generally looks good! Code was easy to read. Cute character lines

porkeypine committed:
Merge branch 'add-gradle-support'

MerlinLim commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,211 @@
+            terminationCommands.add("sayonara");
+
+            while (!terminationCommands.contains(command)) {
+                if (command.equals("list")) {
```
perhaps the use of a switch statement may be better here.

MerlinLim reviewed (commented):
Just some suggestions and noticed a coding violation fo the use of imports. Other than that the code looks great!

porkeypine committed:
Gradle changes

porkeypine committed:
Merge branch 'add-gradle-support' into branch-level-7

porkeypine committed:
Level 7 Save

porkeypine committed:
Level-8 Date (Minimal)

porkeypine committed:
Merge branch 'branch-Level-7'

Fix merge conflicts for Level 7

porkeypine committed:
Merge branch 'branch-Level-8'

Fix merge conflicts for 'branch-Level-8'

porkeypine committed:
Added assertions for reading in data file at the initialisation of Duke

porkeypine committed:
Merge pull request #1 from porkeypine/branch-Assertions

A-Assertions

porkeypine committed:
Added extension C-Update, can now edit task descriptions

porkeypine committed:
Revert "Added extension C-Update, can now edit task descriptions"

This reverts commit e6c1c0df9ee01170a4d6c67c03eaf17b3227f31d.

Pushed to wrong branch

porkeypine committed:
Merge branch 'master' of https://github.com/porkeypine/ip into branch-C-Update

porkeypine committed:
Added extension C-Update, can now edit descriptions of existing tasks

porkeypine committed:
Merge pull request #2 from porkeypine/branch-C-Update

BCD-Extension

porkeypine committed:
A-MoreOOP

porkeypine committed:
A-JUnit

porkeypine committed:
Tweak code to comply with coding standard

Code has been changed to follow the basic and intermediate rules
in SE-EDU Java coding standard.

porkeypine committed:
Fix package and import mistakes for Commands,
improve on coding standards

porkeypine committed:
Add find command and function in Parser, FindCommand, and TaskList

porkeypine committed:
Merge branch 'branch-A-CodingStandard'

Fix merge conflicts in TaskList (un-commented one portion)

porkeypine committed:
Add GUI using javaFX (without fxml)
Add checkstyle files

porkeypine committed:
Add GUI using FXML

porkeypine committed:
Add missed-out resources files

porkeypine committed:
Add JavaDoc for all non-private methods

porkeypine committed:
Add JavaDoc for all classes

porkeypine committed:
Add second constructor to Task and subclasses for initialising tasks
from data file more easily. Correct some checkstyle issues.

porkeypine committed:
Add User Guide

porkeypine committed:
Edit User Guide

porkeypine committed:
Make Ui class methods static

porkeypine committed:
Correct GitHub pages formatting issue

porkeypine committed:
Add TOC to User Guide

porkeypine committed:
Improve user guide

porkeypine committed:
Improve user guide
